### PR TITLE
Basic AUTH with JWT

### PR DESCRIPTION
**Pull request by avelino**
sharing how I imagine endpoint auth token generation
cc @tiaguinho

fixed: #276

### CODE REVIEW

File: cmd/auth.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import (
+	"fmt"
+	"os"
+
+	"github.com/prest/prest/adapters/postgres"
+	"github.com/prest/prest/config"
+	"github.com/spf13/cobra"
+)
+
+var authUpCmd = &cobra.Command{
```
Comment (avelino) [inline cmd/auth.go:12]: command not show in `--help`
```
❯ prestd (avelino/auth) ✘ go run main.go --help
Serve a RESTful API from any PostgreSQL database, start HTTP server

Usage:
  prest [flags]
  prest [command]

Available Commands:
  help        Help about any command
  migrate     Execute migration operations
  version     Print the version number of pREST

Flags:
  -h, --help   help for prest

Use "prest [command] --help" for more information about a command.
```

**Review by avelino — COMMENTED** [review]
we need to document the change, create a topic within the documentation

File: cmd/auth.go (hunk 1 above)
Comment (tiaguinho) [reply cmd/auth.go:12]: It's available inside migrate up.
Do you think will be better if the command was ```auth up``` and ``` auth down```?

Comment (avelino) [reply cmd/auth.go:12]: no, but we need to document
```
go run main.go migrate up auth
go run main.go migrate down auth
```

File: cmd/root.go [hunk 2]
```diff
@@ -55,6 +57,10 @@ func Execute() {
 func MakeHandler() http.Handler {
 	n := middlewares.GetApp()
 	r := router.Get()
+	// if auth is enabled
+	if config.PrestConf.AuthEnabled {
+		r.HandleFunc("/auth", controllers.Auth).Methods("POST")
```
Comment (avelino) [inline cmd/root.go:62]: if jwt has assets we close the endpoint `/auth` being barred by 401 in middleware
```toml
[jwt]
default = true
```
```
❯ prest (avelino/auth) ✘ curl -i -X POST curl -X POST http://127.0.0.1:3000/auth -H "Content-Type: application/json" -d '{"email": "t@avelino.xxx", "password": "123456"}'
curl: (6) Could not resolve host: curl
HTTP/1.1 401 Unauthorized
Content-Type: application/json
X-Content-Type-Options: nosniff
Date: Sat, 31 Oct 2020 14:28:17 GMT
Content-Length: 54

{
	"error": "Required authorization token not found"
}%                                                                                                                                                                                                  ❯ prest (avelino/auth) ✘
```

File: cmd/root.go [hunk 3]
```diff
@@ -55,6 +57,10 @@ func Execute() {
 func MakeHandler() http.Handler {
 	n := middlewares.GetApp()
 	r := router.Get()
+	// if auth is enabled
+	if config.PrestConf.AuthEnabled {
```
Comment (avelino) [inline cmd/root.go:61]: write test if the endpoint is available when active `enabled `

**Review by avelino — COMMENTED** [review]
@tiaguinho can you solve this problem?

```
package github.com/prest/prest/controllers (test)
	imports github.com/prest/prest/middlewares
	imports github.com/prest/prest/controllers: import cycle not allowed in test
FAIL	github.com/prest/prest/controllers [setup failed]
```

**Pull request merged**